tests/integ/modin/index/test_reindex.py

```diff
@@ -0,0 +1,269 @@
+                lambda idx: idx.reindex(list("CEBFGA"), method=method, limit=limit),
+                ValueError,
+                f"limit argument for '{method_str}' method only well-defined if index and target are monotonic",
+            )
+    else:
+        with SqlCounter(query_count=2, join_count=2):
+            native_result = native_idx.reindex(
+                list("CADEFG"), method=method, limit=limit
+            )
+            snow_result = snow_idx.reindex(list("CADEFG"), method=method, limit=limit)
+            assert_reindex_result_equal(snow_result, native_result)
+
+
+@sql_count_checker(query_count=2, join_count=2)
+def test_reindex_index_ordered_index_unordered_new_index():
+    native_idx = native_pd.Index([5, 6, 8])
+    snow_idx = pd.Index(native_idx)
+    native_result = native_idx.reindex([6, 8, 7], method="ffill")
+    snow_result = snow_idx.reindex([6, 8, 7], method="ffill")
+    assert_reindex_result_equal(snow_result, native_result)
+
+
+@pytest.mark.parametrize("limit", [None, 1, 2, 100])
+@pytest.mark.parametrize("method", ["bfill", "backfill", "pad", "ffill"])
+def test_reindex_index_datetime_with_fill(limit, method):
+    query_count = 3 if limit else 2
+    join_count = 2
+    with SqlCounter(query_count=query_count, join_count=join_count):
+        native_date_index = native_pd.date_range("1/1/2010", periods=6, freq="D")
+        snow_date_index = create_datetime_index(
+            pd.date_range("1/1/2010", periods=6, freq="D")
+        )
+        native_result = native_date_index.reindex(
+            native_pd.date_range("12/29/2009", periods=10, freq="D"),
+            method=method,
+            limit=limit,
+        )
+        snow_result = snow_date_index.reindex(
+            pd.date_range("12/29/2009", periods=10, freq="D"),
+            method=method,
+            limit=limit,
+        )
+        assert_reindex_result_equal(snow_result, native_result)
+
+
+@sql_count_checker(query_count=2, join_count=2)
+def test_reindex_index_non_overlapping_index():
+    native_idx = native_pd.Index(list("ABC"))
+    snow_idx = pd.Index(native_idx)
+    native_result = native_idx.reindex(list("EFG"))
+    snow_result = snow_idx.reindex(list("EFG"))
+    assert_reindex_result_equal(snow_result, native_result)
+
+
+@sql_count_checker(query_count=2, join_count=2)
+def test_reindex_index_non_overlapping_datetime_index():
+    native_date_index = native_pd.date_range("1/1/2010", periods=6, freq="D")
+    snow_date_index = create_datetime_index(
+        pd.date_range("1/1/2010", periods=6, freq="D")
+    )
+    native_result = native_date_index.reindex(
+        native_pd.date_range("12/29/2008", periods=10, freq="D")
+    )
+    snow_result = snow_date_index.reindex(
+        pd.date_range("12/29/2008", periods=10, freq="D")
+    )
+    assert_reindex_result_equal(snow_result, native_result)
+
+
+@sql_count_checker(query_count=0)
+def test_reindex_index_non_overlapping_different_types_index_negative():
```

Comment:
does this exception match pandas's?

Reply:
This is an exception on our end since we're trying to join columns of different types - pandas doesn't error out, and instead returns the correct result:
```python
In [3]: pd.date_range("1/1/2010", periods=6, freq="D").reindex(list("CAB"))
Out[3]: (Index(['C', 'A', 'B'], dtype='object'), array([-1, -1, -1]))
```

Reply:
Could you file an issue and link it here?